lib/charms/opensearch/v0/helper_cluster.py

```diff
@@ -97,6 +147,42 @@ def nodes_count_by_role(nodes: List[Node]) -> Dict[str, int]:
 
         return result
 
+    @staticmethod
+    def nodes_by_role(nodes: List[Node]) -> Dict[str, List[Node]]:
+        """Get list of nodes by role."""
+        result = {}
+        for node in nodes:
+            for role in node.roles:
+                if role not in result:
+                    result[role] = []
+
+                result[role].append(node)
+
+        return result
+
+    @staticmethod
+    def nodes(
+        opensearch: OpenSearchDistribution,
+        use_localhost: bool,
+        hosts: Optional[List[str]] = None,
+    ) -> List[Node]:
+        """Get the list of nodes in a cluster."""
+        host: Optional[str] = None  # defaults to current unit ip
+        alt_hosts: Optional[List[str]] = hosts
+        if not use_localhost and hosts:
+            host, alt_hosts = hosts[0], hosts[1:]
+
+        nodes: List[Node] = []
+        if use_localhost or host:
```

Comment:
The API will also be queried if `use_localhost and host is None`. In which case it will pass `host=None` to `opensearch.request()` which will use the current node ip. You can see it [here](https://github.com/canonical/opensearch-operator/blob/main/lib/charms/opensearch/v0/opensearch_distro.py#L289)

The only case where it will not query the API is when: `use_localhost: False - host: None`